### DISCUSSION TITLE
feature idea: Curves by load.

### DISCUSSION THREAD

**Discussion by mrfox-88** [Rem0o/FanControl.Releases#739] [Ideas]
Would it be possible to add the ability to set a curve "sensor" to GPU/CPU load?

Some cards have unreliable temp probes, and temp probes fail. The ability to ramp up the fans as the load increases would be beneficial to these systems and might even be able to increase efficiency by preempting the heat.

**Comment by jmarucha** [comment]
As I read that is not that difficult to implement.
The input is
`System.Diagnostic.PerformanceCounter("Processor Information", "% Processor Utility", "_Total")` for cpu counter
or something bit more complex for GPUs (coz usually there's two of them).

**Comment by Rem0o** [comment]
3 things for you to think about:

1) Have you ever looked at a load graph over time? You'll see why it doesn't make much sense as an input for a curve. Now you might say: let's smooth it out with a filter, like an averaging filter! See point 2.
2) We already have a metric that kinda tracks the smoothed-out load over time: accumulated heat, aka temperature.
3) Does increasing your fan speed decreases the load? 

I could go on and on, but basically, in a linear control system, you want your feedback input to be as closely related to your action point (here fan speed) as possible. Fan speed lowers temperature, not load. Using load instead of temperature would be going away from that principle.
